### PR TITLE
Model with id "undefined" returned by collection.get(obj)

### PR DESCRIPTION
**Pull request by ashokdantham**
@dhritzkiv  The Collection::get method has this odd behaviour when the collection contains a model with the id "undefined": collection.get(obj) will return that model whenever obj.id is undefined. An odd edge case, I know, but I ran into this in a production application... among other things, it means that

new Collection([{id: 'undefined'}, {id: 'foo'}, {id: 'bar'}])
will create a collection with length 1.

What was wrong with the definition of get, which explicitly checked for this case?
return index && (index[query] || index[query[this.mainIndex]])

### CODE REVIEW

File: ampersand-collection.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ var isArray = require('lodash/isArray');
 var bind = require('lodash/bind');
 var assign = require('lodash/assign');
 var slice = [].slice;
+var _ = require('lodash');
```
Comment (dhritzkiv) [inline ampersand-collection.js]: Was this accidentally left in?

File: ampersand-collection.js [hunk 2]
```diff
@@ -167,6 +168,8 @@ assign(Collection.prototype, AmpersandEvents, {
     get: function (query, indexName) {
         if (query == null) return;
         var index = this._indexes[indexName || this.mainIndex];
+        if(index && index[query[this.mainIndex]] && query[this.mainIndex] === undefined)
+        return;
```
Comment (dhritzkiv) [inline ampersand-collection.js]: For style purposes, can you place `return` at the end of the previous line?

File: ampersand-collection.js (hunk 1 above)
Comment (ashokdantham) [reply ampersand-collection.js]: I removed it.

File: ampersand-collection.js (hunk 2 above)
Comment (ashokdantham) [reply ampersand-collection.js]: Added return at the end of previous line.

**Comment by dhritzkiv** [comment]
Hi @ashokdantham. I made a small tweak to the code, which integrated the `query[collectionMainIndex] !== undefined` check into the main return clause. See if that's okay.

---

Leaving this unmerged for now, until one more reviewer approves, or a few weeks pass. Then I'll merge, patch the version, and release.

To reviewers: what was happening was, a model with an id of `undefined` would be returned for any `get` call querying for non-existent elements (including in `set`, when checking that a model didn't already exist).

Basically, `index[query[this.mainIndex]]` would resolve to `index[undefined]`, which is actually equivalent to `index["undefined"]` because `undefined.toString() === "undefined"` (since object keys are stored as strings), which would return the model with an id of "undefined". The new code checks that `query[this.mainIndex]` is `undefined`, so to return `undefined` early, as intended.
